### PR TITLE
Discussion Forum Filter Changes 

### PR DESCRIPTION
**Pull request by rafaykhurram**
Contains filter changes. Look at the most recent 2 commits of this pull request ((I merged Stephen's branch into my branch). Had to add the filters and update the util and route functions for searching through the posts based on the filters

### CODE REVIEW

**Review by mayankshouche — APPROVED** [review]
This looks really good! I only had one small nitpick - when the search doesn't return anything on the discussion forum, it's pushed off a little to the right like so: 
image

On the resources, when the search doesn't work, it's left aligned like this:
image


I think it's just as simple as making sure the Layout element renders underneath the Select, but it's not super important so we can go ahead and merge this if we want.

**Comment by stephen8chang** [comment]
@mayankshouche, could you review the PR one more time? It now contains all code from both Rafay and I's PR, and filters + sorting + integration of the discussion card into the discussion forum page

File: src/components/ListAndFilterDiscussionForums.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+              <Select.Option value="byUpdatedDateDesc">
+                Date Updated (Descending)
+              </Select.Option>
+              <Select.Option value="byNameAsc">Name (Ascending)</Select.Option>
+              <Select.Option value="byNameDesc">Name (Descending)</Select.Option>
+              <Select.Option value="byUpvotesAsc">Upvotes (Ascending)</Select.Option>
+              <Select.Option value="byUpvotesDesc">Upvotes (Descending)</Select.Option>
+            </Select>
+      </Affix>
+      <Layout style={{ padding: '24px 24px 24px' }}>
```
Comment (mayankshouche) [inline src/components/ListAndFilterDiscussionForums.js:142]: We should probably wrap the `<Select />` and `<Layout />` in an Ant Design [Grid](https://ant.design/components/grid/) so they're arranged vertically, not next to each other. Its probably as simple as wrapping the Select and the Layout in their own full size rows, something like this:
```
    <Row>
      <Col span={24}><Select  ... /></Col>
    </Row>
    <Row>
      <Col span={24}><Layout  ... /></Col>
    </Row>
```

**Review by mayankshouche — COMMENTED** [review]
Mostly looks good! Ran into a small bug though. When I click on the "Discussion Forum" header link from a resource page, it puts the current page marker on "organizations" for some reason: 
image.

Also see my comment on ListAndFilterDiscussionForums.js for some UI changes that I think would be pretty beneficial.

**Comment by stephen8chang** [comment]
> Mostly looks good! Ran into a small bug though. When I click on the "Discussion Forum" header link from a resource page, it puts the current page marker on "organizations" for some reason:
 Not sure what the issue is here, the href points to the discussion page
> Also see my comment on ListAndFilterDiscussionForums.js for some UI changes that I think would be pretty beneficial.
For now, I've moved the "Sort by" on the sidebar menu to the left

**Comment by mayankshouche** [comment]
This looks good now, we can probably go ahead and merge unless anyone else had anything to bring up.

**Pull request merged**